### PR TITLE
avoid KeyError on promotion without success published

### PR DESCRIPTION
**Pull request by maorfr**
this error will be caught here to give a useful error message: https://github.com/app-sre/qontract-reconcile/blob/921282ce9bafee2e4413ec050d7d61e272c984b6/reconcile/utils/saasherder/saasherder.py#L1828-L1835

### CODE REVIEW

**Review by patjlm — APPROVED** [review]

File: reconcile/utils/promotion_state.py [hunk 1]
```diff
@@ -95,7 +95,9 @@ def get_promotion_data(
         if use_cache and path_v2 in self._promotion_data_cache:
             return self._promotion_data_cache[path_v2]
 
-        data = self._state.get(path_v2)
+        data = self._state.get(path_v2, None)
+        if not data:
+            return None
```
Comment (fishi0x01) [inline reconcile/utils/promotion_state.py:100]: Should be caught by https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/test/utils/test_promotion_state.py#L66

Comment (fishi0x01) [reply reconcile/utils/promotion_state.py:100]: Apologies, that test is for cached commit shas ... We dont have a test for non-existing key without caching yet. Im happy to take a look tomorrow.

**Review by fishi0x01 — APPROVED** [review]

**Comment by jmelis** [comment]
This LGTM, but I would like to see if there's a way to not raise an error in this situation, as defining pub/subs in a single MR should be something possible IMO

**Comment by jmelis** [comment]
@maorfr but yes, perhaps that's for another MR :)

**Pull request closed without merging**